### PR TITLE
add support to send predefined email template

### PR DESCRIPTION
**Pull request by VitaliiBV**
in MailContent.java deleted </h3> formating tag as `./gradlew install` resulted into failed build because of it.

### CODE REVIEW

File: src/main/java/net/sargue/mailgun/content/MailContent.java [hunk 1]
```diff
@@ -6,7 +6,9 @@
 /**
  * This class is deprecated. Don't use it. See below for instructions on how to
  * migrate current code.
- * <h3>Migration guide</h3>
+ *
```
Comment (sargue) [inline src/main/java/net/sargue/mailgun/content/MailContent.java:9]: Why this change?

File: build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ apply plugin: "jacoco"
 
 group 'net.sargue'
 archivesBaseName = 'mailgun'
-version '1.9.2'
+version '1.9.3'
```
Comment (sargue) [inline build.gradle:12]: Version should be 1.10.0 because it changes the API.

File: src/main/java/net/sargue/mailgun/MailBuilder.java [hunk 3]
```diff
@@ -225,6 +225,26 @@ public MailBuilder content(Body body) {
         return text(body.text()).html(body.html());
     }
 
+    /**
+     * Sets the name of the pre saved template that will be used to be sent by Mailgun.
```
Comment (sargue) [inline src/main/java/net/sargue/mailgun/MailBuilder.java:229]: Please rephrase this line, it's hard to understand.

File: src/main/java/net/sargue/mailgun/MailBuilder.java [hunk 4]
```diff
@@ -225,6 +225,26 @@ public MailBuilder content(Body body) {
...
+     *
+     * @param templateName the name of pre saved template on Mailgun
+     * @return this builder
+     */
+    public MailBuilder template(String templateName){
```
Comment (sargue) [inline src/main/java/net/sargue/mailgun/MailBuilder.java:234]: Please keep existing formatting convetions: space between ) and { at the end of the line.

File: src/main/java/net/sargue/mailgun/MailBuilder.java [hunk 5]
```diff
@@ -225,6 +225,26 @@ public MailBuilder content(Body body) {
...
+        return param("template", templateName);
+    }
+
+    /**
+     * This method adds the param {@code h:X-Mailgun-Variables} to the Mailgun request.
+     *
+     * @param variables custom JSON data to be attached to the message.
+     * @return this builder
+     */
+    public MailBuilder variables(String variables){
```
Comment (sargue) [inline src/main/java/net/sargue/mailgun/MailBuilder.java:244]: Fix formatting.

**Review by sargue — CHANGES_REQUESTED** [review]
Review done. Some questions asked, some changes requested. Awaiting reply.

File: src/main/java/net/sargue/mailgun/content/MailContent.java (hunk 1 above)
Comment (VitaliiBV) [reply src/main/java/net/sargue/mailgun/content/MailContent.java:9]: When i run `./gradlew install` it fails with this error
image

Not sure if that my local issue only or anything I can do about it.

**Comment by VitaliiBV** [comment]
Hi,

Thanks for the review!
Requested changes are done. Answered to your questions in comment.
I will duplicate the answer here as well:


*When i run ./gradlew install it fails with this error* *[image: image]
<https://user-images.githubusercontent.com/63702645/113106266-3cf64480-920b-11eb-8bfc-d9670d40bf2a.png>*

*Not sure if that's my local issue only or anything I can do about it
except for changing the formatting to be without </h3> tag. *

Kind regards

**Comment by sargue** [comment]
Thanks for the changes.

The javadoc error is really weird. First time I see it.

Which JDK are you using to build?

**Comment by VitaliiBV** [comment]
Hi,

Im using Java version „15.0.1“ 2020-10-20.
But I tried Java 8 and it went perfectly well.
Which one do you use?

**Comment by sargue** [comment]
I was using Java 8 but when switching to 11 or 15 I indeed can reproduce the problem.

There are also failing steps. And also *sigh* JFrog is sunsetting bintray/jcenter so I need to switch also those parts.

I guess I need to work a bit on the gradle / build part before merging the PR.

**Comment by sargue** [comment]
Actually, I think it would be better to merge the PR and then work on the upgrade build configuration.

**Review by sargue — APPROVED** [review]

**Comment by VitaliiBV** [comment]
Good morning,

Thank you for reviewing and merging!
Sorry, I think I forgot to add a new tag to PR v1.10.1.
Would you be able to do it?

**Comment by sargue** [comment]
I totally forgot, thanks for reminding me.

---
Sergi Baila

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
